### PR TITLE
Refactor player class so that there is a separate class without sprites

### PR DESCRIPTION
**Pull request by maxmeijer42**
The structure is like this:

There is the DummyPlayer class which has all the functions like PerformAction, CanPerformAction etc. It has some events like OnPlayerAction, OnPlayerLose, OnPlayerWin, OnMoveSmoothly, that a GameObject can bind to to show the appropriate animations. Importantly, DummyPlayer has a Clone function which clones the object so that the AI can try things without affecting the real player.

There is the PlayerGameObject class, which has a DummyPlayer as a member variable. PlayerGameObject extends AnimatedMovingGameObject. It binds to the events of the DummyPlayer for updating the direction of the sprite (now moved to DirectionalGameObject.cs), the smooth movement etc. When it is time to perform the next action the protected property NextAction will be used.

There is the interface ITileFieldPlayer which is what the Tiles use in their methods. DummyPlayer is a ITileFieldPlayer.

There is still this one thing that needs to be fixed with the KeyTile. The KeyTile should not use IsVisited to determine if a key is there. It should instead look at the player's inventory. There can be no game state in the TileField / Tile 's.

### CODE REVIEW

**Comment by maxmeijer42** [comment]
I'll fix the conflicts.

File: MeesGame/Gameplay/GameObjects/Player/ITileFieldPlayer.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        /// </summary>
+        /// <param name="action">The PlayerAction to perform</param>
+        void PerformAction(PlayerAction action);
+
+        /// <summary>
+        /// Makes the player win the game
+        /// </summary>
+        void Win();
+    }
+}
```
Comment (jensj12) [inline MeesGame/Gameplay/GameObjects/Player/ITileFieldPlayer.cs]: Missing newline

**Comment by maxmeijer42** [comment]
Yes, this can. No IsMainPlayer property needed. The PlayerGameObject calls Tile.Visit() in which areas are revealed and tiles are marked visited

**Review by jensj12 — APPROVED** [review]

File: MeesGame/MeesGame.csproj [hunk 2]
```diff
@@ -42,10 +42,15 @@
...
     <Compile Include="Gameplay\GameObjects\Direction.cs" />
+    <Compile Include="Gameplay\GameObjects\DirectionalGameObject.cs" />
+    <Compile Include="Gameplay\GameObjects\Player\DummyPlayer.cs" />
     <Compile Include="Gameplay\GameObjects\KeyColor.cs" />
     <Compile Include="Gameplay\GameObjects\Player\InventoryItem.cs" />
     <Compile Include="Gameplay\GameObjects\Player\AIPlayer.cs" />
     <Compile Include="Gameplay\GameObjects\Player\InventoryItems\InventoryKey.cs" />
+    <Compile Include="Gameplay\GameObjects\Player\ITileFieldPlayer.cs" />
+    <Compile Include="Gameplay\GameObjects\Player\UntimedPlayer.cs" />
+    <Compile Include="Gameplay\Generator\MazeGenerator.cs" />
```
Comment (jensj12) [inline MeesGame/MeesGame.csproj:53]: Merge mistake

**Pull request merged**